### PR TITLE
Write usage tutorial in README

### LINKED ISSUE #68: Expand documentation on what's needed after the base image is built
## Description

We need to have a tutorial in the README explaining how to take a sample repo and compile it using the tool

## Test Plan

* Check that the doc instructions are actually working
* Evaluate if an e2e test could validate automatically

## Documentation Plan

* README.md update

## Release Plan

* Merge into master

## Acceptance Criteria

* [ ] Test plan has been completed
* [ ] Release plan has been completed

Once all above items are checked, this story can be moved to done.

### PR DESCRIPTION
**Pull request by emersonknapp**
Closes #68

Followed through the steps step-by-step and verified that it works. This is the same process that is followed by the e2e tests, the only difference being path names.

Signed-off-by: Emerson Knapp <emerson.b.knapp@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
 NOTE: For trivial text files, as in the example above, you could just as easily create those files fully within the `--custom-setup-script`. However, for binary data such as precompiled libraries, this feature comes to the rescue.
 
 
+## Tutorial
+
+For a new user, this section walks you through a representative use case, step by step.
+
+In this tutorial, we will cross-compile the File Talker tool https://github.com/ros-tooling/file_talker against ROS2 Dashing, to run on an ARM64 Ubuntu system.
```
Comment (thomas-moulard) [inline README.md]: 5:19    warning  Try to avoid using              Google.We           
                  first-person plural like 'we'.

Comment (thomas-moulard) [inline README.md]: 5:19    warning  Try to avoid using              Google.We           
                  first-person plural like 'we'.

File: README.md [hunk 2]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+This workflow can be generalized to any `.repos` file for your project.
```
Comment (thomas-moulard) [inline README.md]: 6:19    warning  'be generalized' may be         write-good.Passive  
                  passive voice. Use active                           
                  voice if you can.

File: README.md [hunk 3]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+
+NOTE: This tutorial currently assumes a Debian-based (including Ubuntu) Linux distribution as the host platform.
```
Comment (thomas-moulard) [inline README.md]: 8:21    warning  'currently' is a weasel word!   write-good.Weasel

File: README.md [hunk 4]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+        url: https://github.com/ros-tooling/file_talker.git
+        version: master
+    ```
+1. Set up the sysroot environment for the cross-compiler to use
+    * `mkdir -p sysroot/qemu-user-static/`
+    * `mkdir -p sysroot/ros_ws/src/`
+    * `cp /usr/bin/qemu-*-static sysroot/qemu-user-static`
+    * `vcs import ros_ws/src < file_talker.repos`
+
+### Running the Cross-Compilation
```
Comment (thomas-moulard) [inline README.md]: 30:5    warning  'Running the                    Google.Headings     
                  Cross-Compilation' should use                       
                  sentence-style capitalization.

File: README.md [hunk 5]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+
+```
+ros2 run cross_compile cross_compile \  # Invoke the cross-compilation tool
+  --sysroot-path $(pwd) \
+  --rosdistro dashing \
+  --arch aarch64 \
+  --os ubuntu
+```
+
+Let's run through the arguments we passed to the script:
```
Comment (thomas-moulard) [inline README.md]: 40:1    warning  Try to avoid using              Google.We           
                  first-person plural like                            
                  'Let's'.

File: README.md [hunk 6]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+
+* `--sysroot-path $(pwd)`
+
+We are pointing the `cross_compile` tool to the absolute path of the directory containing the `sysroot` directory we created.
+You could run the tool from any directory, but in this case we were already in the directory that contained sysroot, hence `$(pwd)`
+
+* `--rosdistro dashing`
+
+Both ROS and ROS2 distros can be specified by name (e.g. kinetic).
```
Comment (thomas-moulard) [inline README.md]: 49:53   error    Use 'for example' instead of    Google.Latin        
                  'e.g.'.

File: README.md [hunk 7]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+
+* `--arch aarch64`
+
+We are targeting the ARMv8 / ARM64 / aarch64 architecture (which are different names for the same thing).
+`cross_compile -h` will print the supported architectures for this option.
+
+* `--os ubuntu`
+
+The target OS is Ubuntu - the version of the OS will be chosen automatically based on the ROS Distro's target OS.
+In our case for ROS2 Dashing, it is 18.04 Bionic Beaver.
```
Comment (thomas-moulard) [inline README.md]: 60:31   warning  'it is' is too wordy            write-good.TooWordy

File: README.md [hunk 8]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+You can double check this for yourself:
+
+```
+$ file ros_ws/install_aarch64/lib/file_talker/file_talker                                                               0s
+ros_ws/install_aarch64/lib/file_talker/file_talker: ELF 64-bit LSB shared object, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-, for GNU/Linux 3.7.0, BuildID[sha1]=02ede8a648dfa6b5b30c03d54c6d87fd9151389e, not stripped
+```
+
+### Using the build on a target platform
+
+Copy `install_aarch64` onto the target system into a location of your choosing. It contains all of the binaries for _your_ workspace.
```
Comment (thomas-moulard) [inline README.md]: 90:93   warning  'all of' is too wordy           write-good.TooWordy

File: README.md [hunk 9]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
+
+If your workspace has any dependencies that are outside the source tree - that is, if `rosdep` had anything to install during the build - then you will still need to install these dependencies on the target system.
+
+```
+# Run this on the target system, which must have rosdep already installed
+# remember `rosdep init`, `rosdep update`, `apt-get update` if you need them
+rosdep install --from-paths install_aarch64/share --ignore-src --rosdistro dashing -y
+```
+
+Now you may use the ROS installation as you normally would
```
Comment (thomas-moulard) [inline README.md]: 100:45  warning  'normally' is a weasel word!    write-good.Weasel

File: README.md [hunk 10]
```diff
@@ -186,6 +186,113 @@ Now, during the sysroot creation process, you should see the contents of `someth
...
++-- src/
+    |-- file_talker
++-- install_aarch64/
+    |-- ...
+```
+
+The created directory `install_aarch64` is the installation of your ROS workspace for your target architecture.
+You can double check this for yourself:
+
+```
```
Comment (thomas-moulard) [inline README.md]:
```suggestion
```sh
```

**Comment by emersonknapp** [comment]
Ok, ran vale linter locally and removed the majority of the warnings, including all of the ones you brought up @thomas-moulard - will merge if all required checks pass
